From e1dee3832125a074ca0358cac48efaa6a8cabea6 Mon Sep 17 00:00:00 2001
From: Buddy Deck <deckb@users.noreply.github.com>
Date: Sat, 18 Apr 2020 10:28:48 -0400
Subject: [PATCH] update build script to not check for submodules

---
 scripts/eosio_build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/eosio_build.sh b/scripts/eosio_build.sh
index 32de8ec0f..fc9e21df7 100755
--- a/scripts/eosio_build.sh
+++ b/scripts/eosio_build.sh
@@ -143,7 +143,7 @@ setup
 execute cd $REPO_ROOT
 
 # Submodules need to be up to date
-ensure-submodules-up-to-date
+#ensure-submodules-up-to-date
 
 # Check if cmake already exists
 ( [[ -z "${CMAKE}" ]] && [[ ! -z $(command -v cmake 2>/dev/null) ]] ) && export CMAKE=$(command -v cmake 2>/dev/null) && export CMAKE_CURRENT_VERSION=$($CMAKE --version | grep -E "cmake version[[:blank:]]*" | sed 's/.*cmake version //g')
-- 
2.24.2 (Apple Git-127)

